### ISSUE TITLE
About the implementation of THP

### ISSUE THREAD

**Issue by zhjcp** [ant-research/EasyTemporalPointProcess#31]
Hello, 

There seems to be a bug in your implementation of THP at line-86 of torch_thp.py:

The current implementation is as follows:
```python
enc_out = self.forward(time_seqs[:, :-1], type_seqs[:, :-1], attention_mask[:, 1:, :-1])
```

The correct implementation should be as follows (right ?):

```python
enc_out = self.forward(time_seqs[:, :-1], type_seqs[:, :-1], attention_mask[:, :-1, :-1])

**Comment by iLampard** [comment]
Hi, 

based on our processing pipeline, the first row of att mask is all "True", see https://ant-research.github.io/EasyTemporalPointProcess/user_guide/dataset.html#an-example 

attention_mask[:, 1:, :-1] =>  its upperleft element is False => which means it can see itself.  

This seems fine to me.  besides, this is the same code as in [ANHP](https://github.com/yangalan123/anhp-andtt/blob/master/anhp/model/xfmr_nhp_fast.py#L138). Let me know if you agree on this.

**Comment by zhjcp** [comment]
Okay, I tested easytpp, and indeed the first row of attention_mask is all True. The current code should be correct, but the design of setting the entire first row to True and then discarding this row is a bit confusing. This approach is different from common designs (refer to `get_attn_key_pad_mask()` and `get_attn_key_pad_mask()` in https://github.com/SimiaoZuo/Transformer-Hawkes-Process/blob/master/transformer/Models.py)."

**Comment by iLampard** [comment]
Hi, 

To avoid such confusion, we will upgrade the repo in the near future, to make it more consistent with huggingface code style.

**Comment by zhjcp** [comment]
Fine !
